bootstrap/secret/secure.go

```diff
@@ -190,3 +197,78 @@ func (p *SecureProvider) DefaultTokenExpiredCallback(expiredToken string) (repla
 
 	return reReadToken, true
 }
+
+// LoadServiceSecrets loads the service secrets from the specified file and stores them in the service's SecretStore
+func (p *SecureProvider) LoadServiceSecrets(secretStoreConfig config.SecretStoreInfo) error {
+
+	contents, err := os.ReadFile(secretStoreConfig.SecretsFile)
+	if err != nil {
+		return fmt.Errorf("seeding secrets failed: %s", err.Error())
+	}
+
+	data, seedingErrs := p.seedSecrets(contents)
+
+	if secretStoreConfig.DisableScrubSecretsFile {
+		p.lc.Infof("Scrubbing of secrets file disable.")
+		return seedingErrs
+	}
+
+	if err := os.WriteFile(secretStoreConfig.SecretsFile, data, 0); err != nil {
+		return fmt.Errorf("seeding secrets failed: unable to overwrite file with secret data removed: %s", err.Error())
+	}
+
+	p.lc.Infof("Scrubbing of secrets file complete.")
+
+	return seedingErrs
+}
+
+func (p *SecureProvider) seedSecrets(contents []byte) ([]byte, error) {
```

Comment:
Would it be cleaner to take a data structure here rather than a byte array?  Or a pointer to the data structure that you modify directly?

Reply:
This is the contents just read from the file, which this func marshals into struct